pkg/proxy/apis/kubeproxyconfig/validation/validation.go

```diff
@@ -141,15 +142,38 @@ func validateKubeProxyConntrackConfiguration(config kubeproxyconfig.KubeProxyCon
 }
 
 func validateProxyMode(mode kubeproxyconfig.ProxyMode, fldPath *field.Path) field.ErrorList {
+	if runtime.GOOS == "windows" {
+		return validateProxyModeWindows(mode, fldPath)
+	}
+
+	return validateProxyModeLinux(mode, fldPath)
+}
+
+func validateProxyModeLinux(mode kubeproxyconfig.ProxyMode, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
+
 	switch mode {
 	case kubeproxyconfig.ProxyModeUserspace:
 	case kubeproxyconfig.ProxyModeIPTables:
 	case kubeproxyconfig.ProxyModeIPVS:
 	case "":
 	default:
 		modes := []string{string(kubeproxyconfig.ProxyModeUserspace), string(kubeproxyconfig.ProxyModeIPTables), string(kubeproxyconfig.ProxyModeIPVS)}
```

Comment:
I won't block this PR, but this is brittle - listing modes twice.  Better would be:

```
var modesLinux = sets.NewString(
    string(kubeproxyconfig.ProxyModeUserspace),
    string(kubeproxyconfig.ProxyModeIPTables),
    string(kubeproxyconfig.ProxyModeIPVS)
)
func validateProxyModeLinux(mode, ...) {
    if mode == "" || modesLinux.Has(mode) {
        return nil
    }
    return field.ErrorList{field.Invalid(fldPath.Child("ProxyMode"), string(mode), errMsg)}
}
```

or something - the key being that the list of modes is spelled out exactly one time.

Reply:
File a PR to fix this #57127